v2/logging.go

```diff
@@ -1,45 +1,23 @@
 package shuttle
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
+	"io"
+	"log/slog"
 )
 
-type Logger interface {
-	Info(s string)
-	Warn(s string)
-	Error(s string)
-}
-
-// SetLoggerFunc sets the function to be used to acquire a logger when go-shuttle logs.
-func SetLoggerFunc(fn func(ctx context.Context) Logger) {
-	getLogger = fn
-}
-
-var getLogger = func(_ context.Context) Logger { return &printLogger{} }
-
-type printLogger struct{}
-
-func (l *printLogger) Info(s string) {
-	fmt.Println(append(append([]any{}, "INFO - ", time.Now().UTC(), " - "), s)...)
-}
-
-func (l *printLogger) Warn(s string) {
-	fmt.Println(append(append([]any{}, "WARN - ", time.Now().UTC(), " - "), s)...)
-}
+// logging is disabled by default
+var log = slog.New(slog.NewTextHandler(io.Discard, nil))
 
-func (l *printLogger) Error(s string) {
-	fmt.Println(append(append([]any{}, "ERROR - ", time.Now().UTC(), " - "), s)...)
+func getLogger() *slog.Logger {
```

Comment:
to keep the same interface, this needs to return a small adapter to slog.Logger to implement the shuttle.Logger interface from L10